src/util/QMagicEnum.hpp

```diff
@@ -0,0 +1,309 @@
+#pragma once
+
+#include <magic_enum/magic_enum.hpp>
+#include <QString>
+#include <QStringView>
+
+namespace qmagicenum {
+
+namespace detail {
```

Comment:
Could you put the detail namespace above the main namespace so the indentation doesn't get messed up?